### ISSUE TITLE
Provide relevant error messages when nodes are added/removed during cluster upgrade

### ISSUE THREAD

**Issue by soumyalj** [rancher/rke#44]
1. Create a k8s cluster using image rancher/k8s:v1.8.2-rancher5 in cluster.yml file. 
./rke_darwin-amd64 cluster up --cluster-file cluster.yml
2. Update the cluster.yml file with  rancher/k8s:v1.8.3-rancher2 for all the components
and add another node 
3. Upgrade the cluster .  Adding another node during upgrade gives an error message as below:

```
soumyas-MBP:rke soumya$ ./rke_darwin-amd64 cluster upgrade --cluster-file cluster.yml
INFO[0000] Upgrading Kubernetes cluster                 
INFO[0000] [state] Found local kube config file, trying to get state from cluster 
INFO[0000] [state] Fetching cluster state from Kubernetes 
INFO[0000] [state] Successfully Fetched cluster state to Kubernetes ConfigMap: cluster-state 
INFO[0000] [certificates] Getting Cluster certificates from Kubernetes 
INFO[0001] [certificates] Successfully fetched Cluster certificates from Kubernetes 
INFO[0001] [ssh] Checking private key                   
INFO[0001] [ssh] Start tunnel for host [soumyarketest1] 
INFO[0001] [ssh] Start tunnel for host [soumyarketest1] 
INFO[0001] [ssh] Start tunnel for host [soumyarketest3] 
INFO[0001] [ssh] Start tunnel for host [soumyarketes2]  
INFO[0001] [ssh] Start tunnel for host [soumyarketest4] 
INFO[0001] [upgrade] All nodes are Ready                
INFO[0001] [upgrade] Upgrading Control Plane Services   
INFO[0001] [controlplane] Upgrading the Controller Plane.. 
INFO[0003] [controlplane] Successfully pulled kube-api image on host [soumyarketest1] 
INFO[0004] [controlplane] Successfully started kube-api container on host [soumyarketest1] 
INFO[0005] [controlplane] Successfully pulled kube-controller image on host [soumyarketest1] 
INFO[0005] [controlplane] Successfully started kube-controller container on host [soumyarketest1] 
INFO[0006] [controlplane] Successfully pulled scheduler image on host [soumyarketest1] 
INFO[0007] [controlplane] Successfully started scheduler container on host [soumyarketest1] 
INFO[0009] [controlplane] Successfully pulled kube-api image on host [soumyarketest3] 
INFO[0009] [controlplane] Successfully started kube-api container on host [soumyarketest3] 
INFO[0011] [controlplane] Successfully pulled kube-controller image on host [soumyarketest3] 
INFO[0011] [controlplane] Successfully started kube-controller container on host [soumyarketest3] 
INFO[0012] [controlplane] Successfully pulled scheduler image on host [soumyarketest3] 
INFO[0013] [controlplane] Successfully started scheduler container on host [soumyarketest3] 
INFO[0013] [controlplane] Successfully upgraded Controller Plane.. 
INFO[0013] [upgrade] Control Plane Services updgraded successfully 
INFO[0013] [upgrade] Upgrading Worker Plane Services    
INFO[0013] [worker] Upgrading Worker Plane..            
INFO[0014] [worker] Successfully pulled kubelet image on host [soumyarketest1] 
INFO[0015] [worker] Successfully started kubelet container on host [soumyarketest1] 
INFO[0016] [worker] Successfully pulled kube-proxy image on host [soumyarketest1] 
INFO[0016] [worker] Successfully started kube-proxy container on host [soumyarketest1] 
INFO[0018] [worker] Successfully pulled kubelet image on host [soumyarketest3] 
INFO[0018] [worker] Successfully started kubelet container on host [soumyarketest3] 
INFO[0019] [worker] Successfully pulled kube-proxy image on host [soumyarketest3] 
INFO[0020] [worker] Successfully started kube-proxy container on host [soumyarketest3] 
INFO[0042] [worker] Successfully pulled kubelet image on host [soumyarketes2] 
INFO[0043] [worker] Successfully started kubelet container on host [soumyarketes2] 
INFO[0044] [worker] Successfully pulled kube-proxy image on host [soumyarketes2] 
INFO[0044] [worker] Successfully started kube-proxy container on host [soumyarketes2] 
FATA[0075] Failed to set cordonded state for node: soumyarketest4 
```
A relevant error message should be provided to inform the user that addition and removal of nodes is not supported.

**Comment by galal-hussein** [comment]
A fix added to raise an error for adding/removing nodes while upgrade

**Comment by galal-hussein** [comment]
The fix can be tested with v0.0.5-dev

**Comment by soumyalj** [comment]
Tested with  rke version v0.0.5-dev with the following steps:
1. Create a k8s cluster using image rancher/k8s:v1.8.3-rancher1 in cluster.yml file.
./rke_darwin-amd64 cluster up --cluster-file cluster.yml
2 Update the cluster.yml file with rancher/k8s:v1.8.3-rancher2 for all the components
and add another node. 
Relevant error message was displayed as below:
```
soumyas-MBP:rke soumya$ ./rke_darwin-amd64 cluster upgrade --cluster-file clusterdigworking.yml
INFO[0000] Upgrading Kubernetes cluster                 
INFO[0000] [state] Found local kube config file, trying to get state from cluster 
INFO[0000] [state] Fetching cluster state from Kubernetes 
INFO[0000] [state] Successfully Fetched cluster state to Kubernetes ConfigMap: cluster-state 
INFO[0000] [certificates] Getting Cluster certificates from Kubernetes 
INFO[0000] [certificates] Successfully fetched Cluster certificates from Kubernetes 
FATA[0000] Adding or removing Worker plane nodes while upgrade is not supported 
```
Also, tried the steps and removed a node too while upgrading. The error message above was displayed as expected.
